File: frontend/src/types/research.ts
/**
 * Research Session, Task, Evidence, and Report TypeScript types.
 */

export type ResearchDepth = 'quick' | 'standard' | 'deep';

export type ResearchStatus =
  | 'queued'
  | 'planning'
  | 'researching'
  | 'analyzing'
  | 'drafting'
  | 'reviewing'
  | 'finalizing'
  | 'completed'
  | 'failed';

export interface ResearchSession {
  id: string;
  workspace_id: string;
  question: string;
  research_depth: ResearchDepth;
  status: ResearchStatus;
  started_at?: string | null;
  completed_at?: string | null;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
}

export interface ResearchCreatePayload {
  question: string;
  research_depth?: ResearchDepth;
}

export interface EvidenceItem {
  id: string;
  claim: string;
  supporting_excerpt: string;
  confidence: number;
  source_id?: string | null;
  source_title?: string;
  source_url?: string | null;
}

export interface ReportItem {
  id: string;
  title: string;
  summary?: string | null;
  content_markdown: string;
  citations?: CitationItem[];
}

export interface CitationItem {
  key: string;
  source_id?: string;
  source_title: string;
  source_url?: string | null;
  supporting_excerpt?: string;
}

File: backend/app/schemas/research.py
"""Research schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ResearchCreate(BaseModel):
    question: str
    research_depth: str = "standard"


class ResearchSessionResponse(BaseModel):
    id: UUID
    workspace_id: UUID
    question: str
    status: str
    research_depth: str
    started_at: datetime | None
    completed_at: datetime | None
    error_message: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class EvidenceResponse(BaseModel):
    id: UUID
    claim: str
    supporting_excerpt: str
    confidence: float
    source_id: UUID | None
    
    model_config = {"from_attributes": True}


class ReportResponse(BaseModel):
    id: UUID
    title: str
    summary: str | None
    content_markdown: str
    
    model_config = {"from_attributes": True}
